src/LiveDevelopment/Agents/XHRShim.js

```diff
@@ -0,0 +1,126 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, forin: true, maxerr: 50, regexp: true */
+/* global Blob, DOMParser */
+(function(global) {
+    "use strict";
+
+    // This script requires access to the transport to send
+    // file requests to the editor as commands
+    var transport = global._Brackets_LiveDev_Transport;
+
+    var XMLHttpRequest = global.XMLHttpRequest;
+    if(!XMLHttpRequest) {
+        return;
+    }
+
+    function sendMessage(msg) {
+        if(!transport || !transport.send) {
+            console.error("[Brackets LiveDev] No transport set");
+            return;
+        }
+
+        transport.send(JSON.stringify(msg));
+    }
+
+    function XMLHttpRequestLiveDev() {
+        var self = new XMLHttpRequest();
+        var requestUrl;
+        var abortCalled = false;
+
+        function decode(data) {
+            return (new global.TextDecoder('utf8')).decode(data);
```

Comment:
You can't rely on this being in the global, since not all browsers implement this.